types/nightwatch/index.d.ts

```diff
@@ -3450,17 +4133,188 @@ export interface ElementCommands {
+     *
+     *
+     * @example
+     * module.exports = {
+     *  'demo Test': function(browser) {
+     *     const resultElement = await browser.getNextSibling('.features-container li:first-child');
+     *
+     *     console.log('next sibling element Id:', resultElement.getId());
+     *   },
+     *
+     */
+    getNextSibling(
+        selector: definition,
+        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<JSON_WEB_OBJECT>) => void,
+    ): this;
+    getNextSibling(
+        using: LocateStrategy,
+        selector: definition,
+        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<JSON_WEB_OBJECT>) => void,
+    ): this;
+
+    /**
+     *  Returns the element immediately preceding the specified one in its parent's child elements list.
+     * The element will be returned as web element JSON object (with an added `.getId()` convenience method).
+     *
+     *
+     * @example
+     * module.exports = {
+     *  'demo Test': function(browser) {
+     *     const resultElement = await browser.getPreviousSibling('.features-container li:second-child');
+     *
+     *     console.log('previous sibling element Id:', resultElement.getId());
+     *   },
+     *
+     * browser.getPreviousSibling('#web-button', function(result) {
+     *
+     *   console.log(result.value)
+     * }})
+     * await browser.getPreviousSibling('#web-button')
+     * await browser.getPreviousSibling({selector: '#web-button', locateStrategy: 'css selector'})
+     *
+     * // with global element():
+     * const formEl = element('form');
+     * const result = await browser.getPreviousSibling(formEl)
+     *
+     * // with Selenium By() locators
+     * // https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/index_exports_By.html
+     * const locator = by.tagName('form');
+     * const result = await browser.getPreviousSibling(locator)
+     *
+     * // with browser.findElement()
+     * const formEl = await browser.findElement('form');
+     * const result = await browser.getPreviousSibling(formEl)
+     *
+     */
+    getPreviousSibling(
+        selector: definition,
+        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<JSON_WEB_OBJECT>) => void,
+    ): this;
+    getPreviousSibling(
+        using: LocateStrategy,
+        selector: definition,
+        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<JSON_WEB_OBJECT>) => void,
+    ): this;
+
+    /**
+     * Returns true or false based on whether the DOM has any child nodes
+     *
+     * @example
+     * module.exports = {
+     *  'demo Test': function(browser) {
+     *     const result = await browser.hasDescendants('.features-container');
+     *
+     *     console.log('true or false:', result);
+     *   },
+     *
+     */
+    hasDescendants(
+        selector: definition,
+        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<boolean>) => void,
+    ): this;
```

Comment:
```ts
const result = await browser.hasDescendants('.features-container');
```
Based on this type definition the result will always be `this` (whatever it is), right? 
So it's not the `boolean` value that would be passed to the callback.
I'm sure it works for nightwatch when it executes the test, but in TypeScript this type is simply incorrect?

As far as I can see this affects _all_ API functions defined here, not only this one.

Reply:
@lloiser, before I started contributing to this, I saw the same problem and wanted to solve this issue. But the case is that, for instance, if I mention the correct type for `hasDescendents`, other chained functions will complain.
For example
```js
browser
.hasDescendents('<selector>')
.setValue('<selector', 'data') // Property 'setValue' does not exist on type 'NightwatchResult<boolean>'.
                              // Property 'setValue' does not exist on type 'false'.ts(2339)
```
As I am still learning about typescript, Can you please guide me on how we can solve this?

Reply:
Maybe something like this could work:

https://www.typescriptlang.org/play?ts=4.7.2#code/C4TwDgpgBAgg7gQwJbAQIwDYQDwBUA0UAagHxQC8UA8gLYp6EBEwAFhAHaNkBkUACgCcA9nQDOEADJIA1jlIBuAFCKk7YBAEAzBAGNoAIWFxxAqAG9FUK1HHAiCDAFcIACnFYdwIQIBcN4AKqAOaEAG4Ozn6iAcEAlH7wyKiYOKxIomFCSAAmJErWUCwIogAiEKJ67NkIaqJuEB5evv6B7CFQOg4YaLrSAPx+LmmifoZCxhqEAuWOGMB+aEJCWDWxFGShWdnxsIgo6FjYw4SLyxA1eYoAvsrFIOw6UJqOD8BIQuxQe8AuaKNGJjWFgKOg+0Sg7CEaTaFCgCG+UDQADpbPYnK5GIwmIxYvlrOkiFsXJDoUFcZZrAB6SlQAACwFEAFoIAAPSCeZkCYSmabARwCdgQbJQcLomwsISzYUkxHQAAGgtCGjlFKs6QAchAlQJiVCWHElKqOmDgIVimUKrD4UlEUiiqVypVqrUXJjyQV0volit2C57RadO6qTT6UzWezgJzuVBefzBcLRc5xZKMNKobKoAqtcqjRrszr-Y7yUbQexwToiqohQAJc2Oq0I5GoiIYrFQN12usVDjOhmunF4tWiL1nGouCvIeO1h0VINWal0hnMtkQDkaaOxgVCkUt5NSiHptDyxU5j2iTXa8eVqddwOGkEmmz5hwAYWvNdvDZtTYgdhb-exWJOxnJ0aj7Dtm3RAD2xxYCAx7MC6jdQcoE9b1zl9cRtVfd9smnAM5ygBdQ2XCMo28GNfzjbdE2gUQJX3GUj0zE8BBVM8Lw0epsIwN9Jw-EDyRuRRnled5PgJIlQj8TYciBG5RM8cTUOHdCx2kxE1PYeSRJeJSPhUzidQ01idKAA

Reply:
Sorry for the late reply. I tried the above solution, but it's not working. I would love to fix this issue: Can you please join https://discord.gg/3Cjkby8a, so that we can discuss the solution in detail.

I think, we can address this change in new PR, as this PR already contains lot of changes